src/app/hero-list/hero-list.component.ts

```diff
@@ -54,8 +44,42 @@ export class HeroListComponent implements OnInit {
   constructor(private heroService: HeroService, private route: ActivatedRoute, private router: Router) { }
 
   ngOnInit(): void {
-    this.route.queryParams.subscribe(res=> {
-      this.getHeroes(parseInt(res.page)).subscribe();
+    this.loading = true;
+    this.route.queryParams
+    .pipe(
+      switchMap(res=> {
+        return this.heroService.getHeroes(res.page, res.search);
+      })
+    )
+    .subscribe((res) => {
```

Comment:
dont name every argument 'res' , try to find better names . in this case probably 'params' and 'heroList'